packages/compass-telemetry/src/provider.ts

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { createServiceLocator } from 'hadron-app-registry';
+import type { TrackFunction } from './track';
+
+const noop = () => {
+  // noop
+};
+
+export function createNoopTrack(): TrackFunction {
+  return noop;
+}
+
+const TelemetryContext = React.createContext<{
+  createTrack(): TrackFunction;
+}>({ createTrack: createNoopTrack });
+
+export const TelemetryProvider = TelemetryContext.Provider;
```

Comment:
Taking into account that track method has dependencies on preferences and logging, you might want to change the interface here to pick them up automatically from context when setting up the service, this is a bit closer to what service providers should do: require the minimum amount of custom logic provided and encapsulating everything else